src/pages/index.tsx

```diff
@@ -66,21 +65,26 @@ const PortalHomePage = ({
         feedback: [],
       };
       setAttendance(prevAttendances => [...prevAttendances, newAttendance]);
+      setCheckinEvent(event);
+      setCheckinModalVisible(true);
     }
   };
 
   useEffect(() => {
     if (checkInResponse) {
       // In dev mode, this runs twice because of reactStrictMode in nextConfig.
       // This will only be run once in prod or deployment.
-      processCheckInResponse(checkInResponse);
-      // Clear the query params without re-triggering getServerSideProps.
-      window.history.replaceState(null, '', config.homeRoute);
+      const event = processCheckInResponse(checkInResponse);
+      if (event) {
+        setCheckinEvent(event);
+        setCheckinModalVisible(true);
+      }
     }
-  }, [checkInResponse]);
+  }, [checkInResponse, user]);
 
   return (
     <div className={styles.page}>
+      <CheckInModal open={checkinModalVisible} event={checkinEvent} onClose={() => {}} />
```

Comment:
you should set `checkinModalVisible` to false in the `onClose` handler. otherwise, if you check into two events without reloading the page, the second event won't show the modal

```suggestion
      <CheckInModal open={checkinModalVisible} event={checkinEvent} onClose={() => setCheckinModalVisible(false)} />
```